packages/edit-widgets/src/blocks/legacy-widget/index.js

```diff
@@ -22,3 +22,61 @@ export const settings = {
 	edit,
 	transforms,
 };
+
+/**
+ * Special factory function created specifically for the legacy-widget block. For every other block, JS module exports
+ * are used for registration. In case of this special block, the return value of the create function is used instead.
+ *
+ * The rationale is that variations of legacy-widgets block are dynamic rather than static - they can only be known once
+ * the editor settings are available.
+ *
+ * @param {Object} editorSettings Current editor settings.
+ * @return {Object} Block object.
+ */
+export const create = ( editorSettings ) => {
+	const legacyWidgets = editorSettings?.availableLegacyWidgets || {};
```

Comment:
Nitpick: let's use `??` here?

```suggestion
	const legacyWidgets = editorSettings?.availableLegacyWidgets ?? {};
```